R/susie_wrapper.R

```diff
@@ -125,6 +121,113 @@ susie_rss_wrapper <- function(z, R, bhat, shat, n = NULL, var_y = NULL, L = 10,
+    }else{
+        LD_extract = LD_data$combined_LD_matrix[allele_flip$target_data_qced$variant_id, allele_flip$target_data_qced$variant_id, drop = FALSE]
+        ref_panel = allele_flip$target_data_qced %>% select("chrom", "pos", "variant_id", "A1", "A2") 
+        sumstat_processed = allele_flip$target_data_qced %>% arrange(pos)
+    }
+    
+    return(input_processed = list(sumstat_processed = sumstat_processed, LD_extract = LD_extract))
+    
+}
+
+
+susie_rss_pipeline = function(z, R, ref_panel, bhat = NULL, shat = NULL, n, L, var_y, QC = TRUE, impute_QC = TRUE, cond_analysis = TRUE, rcond, R2_threshold, max_L, l_step){
+    final_result = list() 
+    single_effect_res = susie_rss_wrapper(z = z, R = LD_extract, bhat = NULL, shat = NULL, L = 1, n = n, var_y = var_y)
+    if (!any(single_effect_res$pip > 0.05)) {
+        message("No variant has a PIP value greater than 0.05 for single effect model. Will return single effect model result.")
+        final_result$single_effect_result = single_effect_res
+    } else {  
+        # 1. Output 1 & 2 (default):  single effect result + susie_rss no QC
+        result_noqc = susie_rss_wrapper(z = z, R = LD_extract, bhat = NULL, shat = NULL, n = n, L = L, var_y = var_y)
+        result_noqc_post = susie_post_processor(result_noqc, data_x = LD_extract, data_y = list(z), mode = "susie_rss")
+        final_result$noqc_result = result_noqc_post
+        final_result$single_effect_result = single_effect_res
+        if(QC){
+            result_qced = susie_rss_qc(z = z, ref_panel = ref_panel, R = LD_extract, n = n, L = L, impute = impute_QC,
+            rcond = rcond, R2_threshold = R2_threshold, max_L = max_L, l_step = l_step, var_y = var_y)
+            var_impute_kept = names(result_qced$qc_impute_result$pip)
+            result_qced_impute_post = susie_post_processor(result_qced$qc_impute_result, data_x = LD_extract[var_impute_kept, var_impute_kept, drop = FALSE], data_y = list(z = result_qced$qc_impute_result$z), mode = "susie_rss")
+            final_result$qc_only_result = result_qced$qc_only_result
+            final_result$qc_impute_result = result_qced_impute_post
+            result_qced$impute_table$chrom = as.numeric(result_qced$impute_table$chrom)
+            final_result$qc_impute_table = result_qced$qc_impute_table
+        }
+        if(cond_analysis){
+            conditional_noqc = susie_rss_wrapper(z = z, R = LD_extract, bhat = NULL, shat = NULL, n = n, L = L, max_iter = 1, var_y = var_y)
+            conditional_qced = susie_rss_qc(z = z, ref_panel = ref_panel, R = LD_extract, n = n, L = L, impute = impute_QC, var_y = var_y,
+            rcond = rcond, R2_threshold = R2_threshold, max_L = max_L, l_step = l_step, max_iter = 1)
+            final_result$cond_noqc = conditional_noqc
+            final_result$cond_qced = conditional_qced
```

Comment:
For all of these results we need post-processed, again, for different coverages and the top_loci table. That is the whole point of summarizing the results in a concise and informative way to the best of our knowledge.